hydra-node/test/Hydra/Model/Payment.hs

```diff
@@ -36,18 +30,6 @@ instance Ord CardanoSigningKey where
    where
     hashOf = verificationKeyHash . getVerificationKey
 
-instance ToJSON CardanoSigningKey where
```

Comment:
These were wrong for two reasons. The first is that weeder (correctly) considers a type class instance unused if the only mention of it is a type family application of one of its associated type, and no actual function on the type class is ever invoked, which was the case for the `Payment` type. The second is that `IsTx` is not well-defined, and should not have any reason to assume JSON instances (or any instances) in its context. These constraints should be mentioned at the point of use, using `ConstraintKinds` to shorthand the constraints rather than incurring these implementations where they may not be needed at all.